irobotclient/configuration_handler.py

```diff
@@ -0,0 +1,159 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import argparse
+import string
+import re
+import os
+import errno
+
+from irobotclient.custom_exceptions import IrobotClientException
+
+
+def _get_command_line_agrs():
```

Comment:
By default, `parser.parse_args` parses the arguments given on the CLI, which are in `sys.argsv`: https://docs.python.org/3.6/library/argparse.html#parsing-arguments
i.e. `parser.parse_args() == parser.parse_args(sys.argv[1:])`. 

Therefore, as it is, `_get_command_line_agrs` essentially has an implicit dependency on the global `sys.argv`, which will make it a pain to test. 

If you make the requirement of the CLI argument list that this function deals with explicit, it will make testing less painful.